### DISCUSSION TITLE
why is OpenedConnectionTracker not pruning closed connections?

### DISCUSSION THREAD

**Discussion by ivankhoosty** [aws/aws-advanced-jdbc-wrapper#993] [Q&A] [answered]
https://github.com/awslabs/aws-advanced-jdbc-wrapper/blob/main/wrapper/src/main/java/software/amazon/jdbc/plugin/OpenedConnectionTracker.java#L196

I am observing a lot of connections being logged by the tracker, but most of them are actually closed. what would be the use case where someone would want to track connection objects until they are garbage collected, rather than closed? 

maybe offer an option for the tracker to remove closed connections from the queue?

thanks

**Comment by ivankhoosty** [comment]
Hi @sergiyvamz  , do you think this should be changed? thanks

**Reply by sergiyvamz** [reply, marked as answer]
Hi @ivankhoosty,

That's definitely not a perfect manner to keep closed and disposed connections in the queue. I've create an internal ticket for this and will keep you informed on the progress. Is the extra memory consumption blocking your development?

Thank you raising this issue!

**Comment by ivankhoosty** [comment]
Hi, thanks, not blocking anything, just making it difficult to keep track
of open and already closed connections in the logs.
